openml/runs/functions.py

```diff
@@ -25,7 +26,31 @@
 # circular imports
 
 
-def run_task(task, model, avoid_duplicate_runs=True, flow_tags=None, seed=None):
+def run_model_on_task(task, model, avoid_duplicate_runs=True, flow_tags=None,
+                      seed=None):
+    flow = sklearn_to_flow(model)
+
+    # returns flow id if the flow exists on the server, False otherwise
+    flow_id = flow_exists(flow.name, flow.external_version)
+
+    if flow_id == False:
```

Comment:
Now we are back to publishing a flow before knowing whether it actually valid (before we first ran the task)?